[DOCS] Add composite runtime fields

lockewritesdocs opened the pull request:
Adds documentation for composite runtime fields that were added in #76287.

Commit:
[DOCS] Add composite runtime fields

elasticmachine commented:
Pinging @elastic/es-docs (Team:Docs)

elasticmachine commented:
Pinging @elastic/es-search (Team:Search)

romseygeek reviewed (commented):
LGTM, thanks @lockewritesdocs 

Is it worth adding `composite` to the list of supported runtime field types at line 152 as well? Maybe with a link to this section? We should also add a note to the index-time script section saying that composite fields are not supported (yet!)

Commit:
Update snippets and tests

Commit:
Add note that composite runtime fields cannot be indexed yet

lockewritesdocs commented:
Thanks Alan! I added `composite` to the list of supported data types in d863483801ffbe4a29e37a1b86cbc949a334a851, and added a note that you cannot currently index a `composite` runtime field in 6ae2b3156d820fcf26625e55094d6ccb6456dd52. If you're 👍 to those changes, I'll merge and backport appropriately.

romseygeek reviewed (approved):
LGTM, thanks @lockewritesdocs!
